### PR TITLE
Remove extra unnecessary allocations

### PR DESCRIPTION
**Pull request by asaadaldien**
- Reuse the same result buffer for scf.for
- Handle subtensor inserts that alise result buffer.

### CODE REVIEW

File: iree/compiler/Conversion/Common/LinalgBufferizePass.cpp [hunk 1]
```diff
@@ -395,6 +410,8 @@ static LogicalResult analyseSubTensorOp(SubTensorOp subTensorOp,
   }
   return analyseSingleOperandResultOp(subTensorOp.source(),
                                       subTensorOp.result(), plan);
+  // plan.unionSets(subTensorOp.source(), subTensorOp.result());
```
Comment (benvanik) [inline iree/compiler/Conversion/Common/LinalgBufferizePass.cpp:413]: intended?

**Review by MaheshRavishankar — APPROVED** [review]
I am surprised this just worked! Lets land this. I am going to dig into this a bit more to see how this actually ended up working :)

File: iree/compiler/Conversion/Common/LinalgBufferizePass.cpp (hunk 1 above)
Comment (asaadaldien) [reply iree/compiler/Conversion/Common/LinalgBufferizePass.cpp:413]: Nop :) thanks for catching this

**Comment by asaadaldien** [comment]
> I am surprised this just worked! Lets land this. I am going to dig into this a bit more to see how this actually ended up working :)

Will see how long this will hold as we move things up to tensors before bufferization, so far on the small scale example I am doing (tile-and-pad, tile, vectorize, bufferize) it seem to work so far :).

**Pull request merged**